_misc/ensure_unicode_literals.py

```diff
@@ -42,24 +42,29 @@ def process_file(path):
 def fix_file(path):
     with open(path, "rb") as fp:
         source = fp.read().decode("utf-8")
-
-    encoding_comment_ctr = ["# -*- coding: utf-8 -*-"]
-
-    def snarf_comment(match):
-        encoding_comment_ctr[0] = match.group(0)
-        return ""
-    source = encoding_comment_regexp.sub(snarf_comment, source)
+        source_lines = source.splitlines()
+
+    need_encoding_comment = True
+    first_non_comment_line_index = 0
+    for line_index, line in enumerate(source_lines):
+        if not line.strip():
+            continue
+        if encoding_comment_regexp.match(line):
+            need_encoding_comment = False
+        if not line.startswith("#"):
+            first_non_comment_line_index = line_index
+            break
```

Comment:
IMHO we should only install encoding comment headers to files that are not all ASCII.